portability-api/src/main/java/org/dataportabilityproject/webapp/StartCopyHandler.java

```diff
@@ -48,51 +47,97 @@
 
-  public StartCopyHandler(ServiceProviderRegistry serviceProviderRegistry, JobDao jobDao,
-      CloudFactory cloudFactory, CryptoHelper cryptoHelper, TokenManager tokenManager) {
+  @Inject
+  StartCopyHandler(ServiceProviderRegistry serviceProviderRegistry, JobDao jobDao,
+      CloudFactory cloudFactory,
+      CryptoHelper cryptoHelper,
+      CommonSettings commonSettings,
+      TokenManager tokenManager) {
     this.serviceProviderRegistry = serviceProviderRegistry;
     this.jobDao = jobDao;
     this.cloudFactory = cloudFactory;
     this.cryptoHelper = cryptoHelper;
+    this.commonSettings = commonSettings;
+    this.tokenManager = tokenManager;
+  }
+
+  /* Test constructor used only for verification of xsrf token flow */
+  @VisibleForTesting
+  StartCopyHandler(TokenManager tokenManager) {
     this.tokenManager = tokenManager;
+    this.serviceProviderRegistry = null;
+    this.jobDao = null;
+    this.cloudFactory = null;
+    this.cryptoHelper = null;
+    this.commonSettings = null;
   }
 
   @Override
   public void handle(HttpExchange exchange) throws IOException {
     Preconditions.checkArgument(
         PortabilityApiUtils.validateRequest(exchange, HttpMethods.POST, "/_/startCopy"));
 
+    String jobId = validateJobId(exchange.getRequestHeaders());
+
+    if (commonSettings.getEncryptedFlow()) {
+      handleWorkerAssignmentFlow(exchange, jobId);
+    } else {
+      handleStartCopyInApi(exchange, jobId);
+    }
+  }
+
+  /**
+   * Validates that the request coming is for a valid job and contains the correct xsrf header
+   */
+  @VisibleForTesting
+  String validateJobId(Headers requestHeaders) {
     String encodedIdCookie = PortabilityApiUtils
-        .getCookie(exchange.getRequestHeaders(), JsonKeys.ID_COOKIE_KEY);
+        .getCookie(requestHeaders, JsonKeys.ID_COOKIE_KEY);
     Preconditions
         .checkArgument(!Strings.isNullOrEmpty(encodedIdCookie), "Encoded Id cookie required");
 
     // Valid job must be present
     String jobId = JobUtils.decodeId(encodedIdCookie);
 
-    // Validate XSRF token is present in request header. strip out any quotation marks that Angular adds and remove whitespace.
-    String token = exchange.getRequestHeaders().getFirst(JsonKeys.XSRF_HEADER).replace("\""," ").trim();
-    Preconditions.checkArgument(!Strings.isNullOrEmpty(token));
-    Preconditions.checkArgument(tokenManager.verifyToken(token));
-
-    if (PortabilityFlags.encryptedFlow()) {
-      handleWorkerAssignmentFlow(exchange, jobId);
-    } else {
-      handleStartCopyInApi(exchange, jobId);
-    }
+    // Validate XSRF token is present in request header. The cookie value might be surrounded by
+    // double quotes which causes the angular cli to also surround the header with double quotes.
+    // Since the value itself may not contain quotes or whitespace, trim off the double quotes by
+    // converting them to whitespace.
+    String tokenHeader = requestHeaders.getFirst(JsonKeys.XSRF_HEADER)
```

Comment:
Should this be a utility method in a utils class with the javadoc explaining it?